libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/ide/IdeMultiplatformImport.kt

```diff
@@ -149,31 +152,67 @@ interface IdeMultiplatformImport {
     }
 
     /**
-     * Any [IdeDependencyResolver] has to be registered specifying a certain resolution level.
-     * Generally, all resolvers registered in a given resolution level will work collaboratively, meaning the dependency resolution
-     * result is the aggregation of all resolvers running.
+     * Indicating the 'priority' of a given [IdeDependencyResolver] or [IdeAdditionalArtifactResolver]:
+     * When resolving dependencies, only the highest priority resolver will be selected and executed.
+     * If there are multiple resolvers with the same [Priority] registered, then all of them will be executed
+     * and the result will be merged.
      *
-     * However, only the resolvers in the highest resolution result will run e.g.
-     * If resolvers with level [Overwrite] are found, then only those will contribute to the dependency resolution.
-     * Otherwise, all [Default] resolvers will run.
+     * By 'default' resolvers are registered using [Priority.normal].
+     * The Kotlin Gradle Plugin is not expected to register resolvers outside the [normal] or [high] priorities
+     * External Kotlin Target maintainers _(such as Google)_ can therefore use [veryHigh] to overwrite all resolvers
+     * from the Kotlin Gradle plugin or [low] to only register a resolver if there is nothing available by default
+     * from the Kotlin Gradle plugin.
      */
     @ExternalKotlinTargetApi
-    enum class DependencyResolutionLevel {
-        Default, Overwrite
+    class Priority(val value: Int) : Comparable<Priority> {
```

Comment:
Is there a reason why this is very fine grained rather than just an enum? I understand the need for having the four phases mentioned below, but why do you want to give such flexibility?

Reply:
Yes! There are two reasons for this change (maybe three)

1): You see that KGP already also overwrites some resolvers with a constraint and the Level.Overwrite. Such overwrites therefore cannot again be overwritten by you folks

2): The API was not intuitive enough. Everybody who stumbled over it had to ask for how it works. I really hope that Priority is just simpler to understand 

3): Enums in public API are much more complicated. How could we have added a value safely? This could have been a rather complicated process!

Reply:
I'm worried about that there is no way for a target to know if its resolvers end up running or not. Also if a plugin wants to match the highest priority that currently exists and just contribute to the resolution without completely overriding it.

Reply:
If we know about uses cases where its hard for a third party to sync with AGP and KGP, then we could start designing new APIs to make this cases easier!